cmd/uplink/cmd/cp.go

```diff
@@ -100,24 +85,19 @@ func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.
 		bar.Finish()
 	}
 
-	fmt.Printf("Created %s\n", destObj)
+	fmt.Printf("Created %s\n", destObj.String())
 
 	return nil
 }
 
 // download downloads s3 compatible object args[0] to args[1] on local machine
-func download(ctx context.Context, bs buckets.Store, srcObj *url.URL, destFile string) error {
-	var err error
-	if srcObj.Scheme == "" {
-		return fmt.Errorf("Invalid source")
-	}
-
-	o, err := bs.GetObjectStore(ctx, srcObj.Host)
+func download(ctx context.Context, bs buckets.Store, srcObj fpath.FPath, destObj fpath.FPath) error {
```

Comment:
Let's add a guard here for checking for checking:
```
if srcFile.IsLocal() || !destObj.IsLocal() { return some error }
```

Reply:
Done.